### PR TITLE
Refactor inventory plugin connection handling

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Further refactors inventory plugin connection handling:

- Expands AWSPluginBase client() and resource() to allow overriding parameters
-  renames `_boto3_conn` to `all_clients` to better describe what it's doing (`_boto3_conn` is used elswhere for *singluar* clients)
- assumes the IAM role *once*
- updates plugin parameters to match other plugins/modules
- updates RDS description wrapper to avoid reordering arguments
- avoids maintaining a full inventory scoped rewrite of client/resource code
- avoids maintaining a full inventory scoped handling of boto3/botocore version handling

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/inventory/aws_ec2.py
plugins/inventory/aws_rds.py
plugins/plugin_utils/base.py
plugins/plugin_utils/inventory.py

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: plugins/doc_fragments/assume_role.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class ModuleDocFragment:
+    # Note: If you're updating MODULES, PLUGINS probably needs updating too.
+
+    # Formatted for Modules
+    # - modules don't support 'env'
+    MODULES = r"""
+options: {}
+"""
+
+    # Formatted for non-module plugins
```
Comment (alinabuzachis) [inline plugins/doc_fragments/assume_role.py:15]: Is it worth adding a note in the README about this doc_fragment?

Comment (tremble) [reply plugins/doc_fragments/assume_role.py:15]: The scope is currently limited to inventory plugins, so I wasn't going to add one just yet.

File: plugins/doc_fragments/assume_role.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2015, Ansible, Inc
```
Comment (alinabuzachis) [inline plugins/doc_fragments/assume_role.py:2]: Should it be 2015 or 2022?

**Review by alinabuzachis — APPROVED** [review]

File: plugins/doc_fragments/assume_role.py (hunk 2 above)
Comment (tremble) [inline plugins/doc_fragments/assume_role.py:2]:
```suggestion
# Copyright: (c) 2022, Ansible, Inc
```

**Pull request merged**